src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,208 @@
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Encapsulates methods that handle and interpret user input.
+ */
+public class Parser {
+
+    /** The TaskList that will be altered based on user input */
+    private TaskList taskList;
+
+    /**
+     * Enum that holds information about acting on a task, not including actions that changes the size of the TaskList.
+     * Task actions include deleting a task (DELETE) and marking a Task as done (DONE).
+     */
+    private enum TaskAction {
+        DELETE("delete", 7, "deleted"),
+        DONE("done", 5, "marked as done");
+
+        /** Name of the task action */
+        private String name;
+
+        /** Index to substring the input so as to remove the first word (command word) */
+        private int substringIndex;
+
+        /** Message fragment when the action is successfully completed */
+        private String successMessage;
+
+        /**
+         * Constructs a TaskAction.
+         *
+         * @param name Name of the TaskAction.
+         * @param substringIndex Index to substring the input so as to remove the first word (command word).
+         * @param successMessage Message fragment when the action is successfully completed.
+         */
+        TaskAction(String name, int substringIndex, String successMessage) {
+            this.name = name;
+            this.substringIndex = substringIndex;
+            this.successMessage = successMessage;
+        }
+    }
+
+    /**
+     * Constructs a Parser instance that acts on the given TaskList.
+     *
+     * @param taskList TaskList that the Parser will alter or act on.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Takes user input to interpret and acts on the TaskList appropriately.
+     *
+     * @param input Input string from the user.
+     */
+    public boolean handleInput(String input) {
+        Ui.printSingleDivider();
+        if (input.equals("bye")) {
+
+            return false;
+
+        } else if (input.equals("list")) {
+
+            System.out.println("Output: This is your current list!\n");
+            taskList.printList();
+
+        } else if (input.startsWith("find")) {
+
+            findTask(input);
+
+        } else if (input.startsWith("done")) {
+
+            alterTask(input, Parser.TaskAction.DONE);
+
+        } else if (input.startsWith("delete")) {
+
+            // If successfully deleted task, then print current number of tasks.
+            if (alterTask(input, Parser.TaskAction.DELETE)) {
+                taskList.printNumberOfTasks();
+            }
+
+        } else {
+            vetoTask(input);
+        }
+
+        Ui.printDoubleDivider();
+        return true;
+    }
+
+    private void findTask(String input) {
+        try {
+            System.out.println("Output: These tasks have descriptions that contain the phrase '"
+                    + input.substring(5) + "'!\n");
```

Comment:
Should this magic number be moved to top by using named constant? 🤔

Reply:
Done!